### PR TITLE
Various simplifications suggested by Clippy.

### PR DESCRIPTION
**Pull request by ltratt**
Various fixes suggested by Clippy.

### CODE REVIEW

File: lrpar/src/lib/ctbuilder.rs [hunk 1]
```diff
@@ -363,7 +363,9 @@ where
                         // Iterate over all $-arguments and replace them with their respective
                         // element from the argument vector (e.g. $1 is replaced by args[0]). At
                         // the same time extract &str from tokens and actiontype from nonterminals.
-                        outs.push_str(&format!("fn {prefix}action_{}({prefix}input: &str, {prefix}args: &Vec<AStackType<{actiont}, {}>>) -> {actiont} {{\n",
+                        outs.push_str(&format!("#[allow(clippy::let_and_return)]
```
Comment (ptersilie) [inline lrpar/src/lib/ctbuilder.rs]: What is this `let_and_return` for? I can't see it being used here.

Comment (ltratt) [reply lrpar/src/lib/ctbuilder.rs]: Basically Clippy complains about generated functions that look like:

```rust
fn f() {
    let x = ...;
    x
}
```

and this turns that warning off.

**Comment by ptersilie** [comment]
One comment, rest looks fine. I was suprised that clippy even suggests stuff within strings, or did you edit that manually?

File: lrpar/src/lib/ctbuilder.rs (hunk 1 above)
Comment (ptersilie) [reply lrpar/src/lib/ctbuilder.rs]: I see.

**Comment by ltratt** [comment]
Most of the string stuff was Clippy. One thing was me reformatting stuff so that I could understand an error in generated code. [It seemed worth keeping.]

**Comment by ptersilie** [comment]
Looks like this needs a rebase first.

**Comment by ltratt** [comment]
Rebased.

**Pull request merged**